packages/hardhat-core/src/internal/core/providers/accounts.ts

```diff
@@ -117,11 +117,7 @@ export class LocalAccountsProvider extends ProviderWrapperWithChainId {
       // if we don't manage the address, the method is forwarded
       const privateKey = this._getPrivateKeyForAddressOrNull(address);
       if (privateKey !== null) {
-        return signTypedData({
-          privateKey,
-          version: SignTypedDataVersion.V4,
-          data: typedMessage,
-        });
+        return signTyped(typedMessage, privateKey, false);
```

Comment:
Are we sure we want a false here?

Reply:
I believe we do. Using `extraEntropy` will make the signature different every time, which not only messes up our tests but could also affect user tests and end up being a breaking change.

Reply:
Let's merge with `false`. I created a follow up.